File: src/main/java/me/rohank05/commands/music/LoopCommand.java
package me.rohank05.commands.music;

import me.rohank05.utilities.command.CommandPermissionCheck;
import me.rohank05.utilities.command.ICommand;
import me.rohank05.utilities.music.PlayerManager;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;

import java.util.Objects;

public class LoopCommand implements ICommand {
    private final PlayerManager playerManager;
    public LoopCommand(PlayerManager playerManager){
        this.playerManager = playerManager;
    }
    @Override
    public void run(SlashCommandInteractionEvent event) {
        if (!CommandPermissionCheck.checkBasePermission(event)) return;
        if (!CommandPermissionCheck.checkPermission(event, this.playerManager)) return;
        String LoopSubcommand = Objects.requireNonNull(event.getSubcommandName()).toLowerCase();
        playerManager.getGuildMusicManager(Objects.requireNonNull(event.getGuild())).trackManager.loop = LoopSubcommand;
        EmbedBuilder embedBuilder = new EmbedBuilder().setColor(16760143);
        switch (LoopSubcommand) {
            case "off" -> embedBuilder.setDescription("Loop Off!!");
            case "queue" -> embedBuilder.setDescription("Looping Queue!");
            case "track" -> embedBuilder.setDescription("Looping Track!");
        }
        event.getInteraction().getHook().sendMessageEmbeds(embedBuilder.build()).queue();
    }

    @Override
    public String getName() {
        return "loop";
    }

    @Override
    public String getDescription() {
        return "loop the queue/track";
    }
}

File: src/main/kotlin/com/therohankumar/commands/LoopCommand.kt
package com.therohankumar.commands

import com.therohankumar.audio.LoopMode
import com.therohankumar.audio.MusicManagerRegistry
import com.therohankumar.commands.util.CommandUtil
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.Commands
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData
import net.dv8tion.jda.api.interactions.commands.build.OptionData
import net.dv8tion.jda.api.interactions.commands.Command as JDACommand
import org.springframework.stereotype.Component

@Component
class LoopCommand(private val registry: MusicManagerRegistry) : SlashCommand {

    override val name = "loop"

    override fun build(): SlashCommandData = Commands.slash(name, "Set the loop mode")
        .addOptions(
            OptionData(OptionType.STRING, "mode", "Loop mode", true)
                .addChoices(
                    JDACommand.Choice("Off", "off"),
                    JDACommand.Choice("Track", "track"),
                    JDACommand.Choice("Queue", "queue")
                )
        )

    override fun execute(event: SlashCommandInteractionEvent) {
        val manager = CommandUtil.requireSameVoice(event, registry) ?: return
        val mode = when (event.getOption("mode")!!.asString) {
            "track" -> LoopMode.TRACK
            "queue" -> LoopMode.QUEUE
            else -> LoopMode.OFF
        }
        manager.scheduler.loopMode = mode
        val label = when (mode) {
            LoopMode.OFF -> "🔁 Loop disabled"
            LoopMode.TRACK -> "🔂 Looping current track"
            LoopMode.QUEUE -> "🔁 Looping entire queue"
        }
        event.replyEmbeds(CommandUtil.successEmbed("Loop", label)).queue()
    }
}
